src/v0/util/googleUtils/index.js

```diff
@@ -1,36 +1,97 @@
+ * ref : https://developers.google.com/google-ads/api/rest/reference/rest/v16/Consent
  */
 
-const populateConsentForGoogleDestinations = (config) => {
+const populateConsentFromConfig = (config) => {
   const consent = {};
 
   if (config?.userDataConsent) {
     if (GOOGLE_ALLOWED_CONSENT_STATUS.includes(config.userDataConsent)) {
       consent.adUserData = config.userDataConsent;
     } else {
-      consent.adUserData = 'UNKNOWN';
+      consent.adUserData = UNKNOWN_CONSENT;
     }
   } else {
-    consent.adUserData = 'UNSPECIFIED';
+    consent.adUserData = UNSPECIFIED_CONSENT;
   }
 
   if (config?.personalizationConsent) {
     if (GOOGLE_ALLOWED_CONSENT_STATUS.includes(config.personalizationConsent)) {
       consent.adPersonalization = config.personalizationConsent;
     } else {
-      consent.adPersonalization = 'UNKNOWN';
+      consent.adPersonalization = UNKNOWN_CONSENT;
     }
   } else {
-    consent.adPersonalization = 'UNSPECIFIED';
+    consent.adPersonalization = UNSPECIFIED_CONSENT;
   }
   return consent;
 };
 
-module.exports = { populateConsentForGoogleDestinations };
+/**
+ * Populates the consent object based on the provided properties.
+ *
+ * @param {object} properties - message.properties containing properties related to consent.
+ * @returns {object} - An object containing consent information.
+ *  * ref :
+ * 1) For click conversion :
+ *  a) https://developers.google.com/google-ads/api/rest/reference/rest/v16/customers/uploadClickConversions#ClickConversion
+ *  b) https://developers.google.com/google-ads/api/reference/rpc/v16/ClickConversion#consent
+ * 2) For Call conversion :
+ *  a) https://developers.google.com/google-ads/api/rest/reference/rest/v16/customers/uploadCallConversions#CallConversion
+ *  b) https://developers.google.com/google-ads/api/reference/rpc/v16/CallConversion#consent
+ * 3) For Store sales conversion :
+ *  a) https://developers.google.com/google-ads/api/reference/rpc/v16/UserData
+ *  b) https://developers.google.com/google-ads/api/reference/rpc/v16/UserData#consent
+ */
+
+const populateConsentForGAOC = (message, conversionType, destConfig) => {
+  const integrationObj =
+    conversionType === 'store'
+      ? {}
+      : getIntegrationsObj(message, 'GOOGLE_ADWORDS_OFFLINE_CONVERSIONS') || {};
+  const consents = integrationObj?.consents || {};
+
+  const defaultConsentBlock = populateConsentFromConfig(destConfig);
+
+  // Define a function to process consent based on type
+  const processConsent = (consentType) => {
+    if (!consents[consentType]) {
+      return defaultConsentBlock[consentType] || UNSPECIFIED_CONSENT;
+    }
+    if (GOOGLE_ALLOWED_CONSENT_STATUS.includes(consents[consentType])) {
+      return consents[consentType];
+    }
+    return defaultConsentBlock[consentType] || UNKNOWN_CONSENT;
+  };
+
+  // Common consent fields to process
+  const consentFields = ['adUserData', 'adPersonalization'];
```

Comment:
Let's keep this context outside of this function and make the caller responsible for sending this data.

Reply:
done